### PR TITLE
XKCD Content Provider support (take 2)

### PR DESCRIPTION
**Pull request by ctolkien**
Heavily based off the existing Bash content provider, although no longer quite so much @khellang :)

Looks like this

Has alt text as the title of the image, so no spoilers.

### CODE REVIEW

File: JabbR/ContentProviders/XkcdContentProvider.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            return ExtractFromResponse(request).Then(pageInfo =>
+            {
+                if (pageInfo == null)
+                {
+                    return null;
+                }
+
+                return new ContentProviderResult
+                {
+                    Content = String.Format(ContentFormat, request.RequestUri.ToString(),  pageInfo.ImageUrl, pageInfo.Title, pageInfo.Description),
```
Comment (hnrkndrssn) [inline JabbR/ContentProviders/XkcdContentProvider.cs]: You're passing 4 parameters, but your only using 3(?) of them in the actual `ContentFormat` string...is there are reason for this?

**Comment by hnrkndrssn** [comment]
This works:
http://xkcd.com/13/

This does not:
http://xkcd.com/13

image

Should we make it check, and append a '/', if the URL doesn't have it?

File: JabbR/ContentProviders/XkcdContentProvider.cs (hunk 1 above)
Comment (ctolkien) [inline JabbR/ContentProviders/XkcdContentProvider.cs]: Good catch, this is a holdover from adjusting what the output looked like.

**Comment by ctolkien** [comment]
@alfhenrik  I enforced the ending trail slash as a.) the website 301's to having the trailing slash (so if you copy pasta from the website, it will have it) and I wasn't sure if the 301 would be followed.

**Comment by hnrkndrssn** [comment]
Yea, I guess you'd be copying it from the website, not manually typing it in like I did :grin:

**Comment by hnrkndrssn** [comment]
Looks good to me :+1: 

@davidfowl @cmaish wanna chime in?

**Comment by khellang** [comment]
It would be nice to allow urls without the trailing slash, after all, xkcd.com allows it :smile:

**Comment by ctolkien** [comment]
Now works with or w/o slash. Seems redirects are followed, so we don't need to append it at our end. 

image

**Comment by khellang** [comment]
:metal:

**Pull request merged**